include/miral/miral/decoration_basic_manager.h

```diff
@@ -14,30 +14,42 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#ifndef MIR_SHELL_DECORATION_DECORATION_H_
-#define MIR_SHELL_DECORATION_DECORATION_H_
+#ifndef MIRAL_DECORATION_BASIC_MANAGER_H
+#define MIRAL_DECORATION_BASIC_MANAGER_H
+
+#include "miral/decoration.h"
+#include <memory>
 
 namespace mir
 {
+class Server;
 namespace shell
 {
+class Shell;
 namespace decoration
 {
-/// Draws window decorations and provides basic move, resize, close, etc functionality
-class Decoration
+class Decoration;
+}
+}
+namespace scene
 {
-public:
-    Decoration() = default;
-    virtual ~Decoration() = default;
+class Surface;
+}
+}
 
-    virtual void set_scale(float new_scale) = 0;
+namespace miral
+{
+class DecorationManagerAdapter;
+class DecorationBasicManager
+{
+public:
+    DecorationBasicManager(mir::Server&, Decoration::DecorationBuilder&&);
 
+    auto to_adapter() -> std::shared_ptr<DecorationManagerAdapter>;
 private:
-    Decoration(Decoration const&) = delete;
-    Decoration& operator=(Decoration const&) = delete;
+    struct Self;
+    std::shared_ptr<Self> self;
 };
```

Comment:
There doesn't seem to be much purpose to this class. It takes "stuff" in its constructor and returns "other stuff" from `to_adapter()` - that's pretty much a function in disguise.

Reply:
It now does a bunch of stuff before decoration creation (creating the window), and a bunch of stuff after to hook up the decoration adapter to some dependencies from the server.